amundsen_application/static/js/components/common/BadgeList/index.tsx

```diff
@@ -4,36 +4,122 @@
+  style: BadgeStyle,
+  displayName: string,
+  action: any;
+}
+
+export type BadgeListProps = ListProps & DispatchFromProps;
+
+
+const StaticBadge: React.FC<BadgeStyleConfig> = ({ style, displayName }: BadgeStyleConfig) => {
   return (
-    <span className="badge-list">
-      {badges.map((badge, index) => {
-        let badgeConfig;
-        // search badges with just name
-        if (badge.tag_name) {
-          badgeConfig = getBadgeConfig(badge.tag_name);
-        }
-        // metadata badges with name and category
-        else if (badge.badge_name) {
-          badgeConfig = getBadgeConfig(badge.badge_name);
+    <span className={`static-badge flag label label-${style}`}>
+      <div className={`badge-overlay-${style}`}>
+        {displayName}
+      </div>
+    </span>);
+};
+
+const ActionableBadge: React.FC<ActionableBadgeProps> = ({ style, displayName, action }: ActionableBadgeProps) => {
+  return (
+    <span className="actionable-badge" onClick={action}>
+      <StaticBadge
+        style={style}
+        displayName={displayName}
+      />
+    </span >
+  );
+};
+
+export class BadgeList extends React.Component<BadgeListProps> {
+  render() {
+    return (
+      <span className="badge-list">
+        {this.props.badges.map((badge, index) => {
+          let badgeConfig;
+          let originalBadgeName;
+          // search badges with just name
+          if (badge.tag_name) {
+            originalBadgeName = badge.tag_name;
+            badgeConfig = getBadgeConfig(badge.tag_name);
+          }
+          // metadata badges with name and category
+          else if (badge.badge_name) {
+            originalBadgeName = badge.badge_name;
+            badgeConfig = getBadgeConfig(badge.badge_name);
+            if (badge.category === COLUMN_BADGE_CATEGORY) {
+              return (<StaticBadge
+                style={badgeConfig.style}
+                displayName={badgeConfig.displayName}
+                key={`badge-${index}`}
+              />);
+            }
+          }
+          if (badge.category !== COLUMN_BADGE_CATEGORY) {
+            let onClick = (e) => {
+              const badgeText = originalBadgeName;
+              logClick(e, {
+                target_type: 'badge',
+                label: badgeText,
+              });
+              this.props.searchBadge(convertText(badgeText, CaseType.LOWER_CASE));
+            };
+            return (<ActionableBadge
+              displayName={badgeConfig.displayName}
+              style={badgeConfig.style}
+              action={onClick}
+              key={`badge-${index}`}
+            />);
+          }
+
         }
-        return (
-          <ClickableBadge
-            text={badgeConfig.displayName}
-            labelStyle={badgeConfig.style}
-            key={`badge-${index}`}
-          />
-        );
-      })}
-    </span>
+        )}
+      </span>);
+  }
+}
+
+export const mapDispatchToProps = (dispatch: any) => {
```

Comment:
I would love us to push this still upwards one more level, so BadgeList and Badges are presentational components.

Not urgent to do, but this is definitely a step on the right direction!

Reply:
I agree, I'll refactor it in a separate PR when i finalize the badges work